BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -124,24 +144,49 @@ uint8_t BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
  * @param   data : pointer to store data that was received. Must be 8bytes or bigger.
  * @return  0 if nothing was received so ignore id and data that was received. Any other value indicates data was received and stored.
  */
-uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
-    if(RxFlag){
+uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) { //originally returns uint8_t
+
+    
+    //if(RxFlag){
```

Comment:
what was `RxFlag` being used for, and is it safe to remove it?

Reply:
I reincluded this and modified the hardware interrupt method so it would set this flag to "True" whenever it's called.